### DISCUSSION TITLE
Advanced SSR docs confusion

### DISCUSSION THREAD

**Discussion by sergical** [TanStack/query#7367] [General]
https://tanstack.com/query/latest/docs/framework/react/guides/advanced-ssr

The code example imports `useState` and `use client` but the actual code seems to be checking for window and then not using useState at all - would this be a server rendered provider without the need for those two (import and `use client`)?

**Comment by TkDodo** [comment]
yeah we don't need the `useState` there anymore. file a PR?

**Reply by sergical** [reply]
Will do, just wanted to double check!

**Reply by sergical** [reply]
We'd still need the `use client` though? If I remove it, I get the following:

```
- error node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js (2102:12) @ resolveModelToJSON
- error Error: Functions cannot be passed directly to Client Components unless you explicitly expose it by marking it with "use server".
  {exclude: undefined, format: function, type: ...}
```

**Reply by TkDodo** [reply]
yes because the QueryClientProvider uses context under the hood

**Reply by sergical** [reply]
https://github.com/TanStack/query/pull/7435
